wordpress-mobile/WordPress-Android
# WP Stories integration - add Tracks (part 1)

This PR builds on top of  #12068 

This PR adds tracking for the WP Stories feature. Guided by looking into all the `// TODO WPSTORIES add TRACKS` comments to think and replace by proper behavior.

### Sequence of events
For a successful Story save and upload:
- trackEditorCreatedPost -> `EDITOR_CREATED_POST` this is when the StoryComposerActivity is shown for a new Story
- trackStorySaveResultEvent -> `STORY_SAVE_SUCCESSFUL`  (`STORY_SAVE_ERROR` if error)
- editorTracker.trackAddMediaFromDevice (`EDITOR_ADDED_VIDEO_VIA_DEVICE_LIBRARY` or `EDITOR_ADDED_PHOTO_VIA_DEVICE_LIBRARY`)
- trackSavePostAnalytics -> N/A (doesn't track anything because for a `PUBLISHED` Post it lets `PostUploadHandler` handle it)
- trackStoryPostSavedEvent -> `STORY_POST_SAVE_LOCALLY` if no network / `STORY_POST_SAVE_REMOTELY` in case it was correctly enqueued to be uploaded to the site. Uses `frame_qty` as a property to indicate the amount of frames this Story has.
- (PostUploadHandler) -> `EDITOR_PUBLISHED_POST` once it's correctly saved on the remote site.


### Changes in this PR
- de83be9 when a StoryResultEvent is received on the bridge, `trackStorySaveResultEvent` is called and tracks either `Stat.STORY_SAVE_SUCCESSFUL` or `Stat.STORY_SAVE_ERROR` with site details and the `isRetry` flag (boolean) to indicate whether this is a save retry by the user.
- 8ad12ec2bf adds STORY_POST_SAVE_LOCALLY and STORY_POST_SAVE_REMOTE events, which area used in the bridge as the last step (if there's network it's going to be saved remotely, if not, it's saved locally)
- d6de7a7317 tracks the specific case of touching the action label `MANAGE` on the error snackbar, instead of counting that as a error Notification tap via `ARG_NOTIFICATION_TYPE`.
-  d223dd4d0f9536b8715e3c636cacb4ec147904b8 moved `trackEditorCreatedPost` method from EditPostActivity to AnalyticsUtils so it's used both in EditPostActivity and StoryComposerActivity. This will track creation of new posts (while at it, also wrote tiny fix in https://github.com/wordpress-mobile/WordPress-Android/pull/12081 included in this PR as well)
- 29273bd includes the `post_format` property when adding postType property to tracks. This is of particular interest for Story posts given the only way to tell one another (from a regular Post) is by looking into this value (for Stories, it's `"wpstory"`)
- 4b1eaf4752 tracks the specific case for users tapping on the camera FAB in the media picker, signaling they want to open the StoryComposerActivity capture mode
- 0f9de40c43 removed `// TODO WPSTORIES add TRACKS` comment there given it's not needed: when picking media from the picker, tracks will be recorded in `trackAddMediaEvent`  in `AddExistingMediaToPostUseCase`.
- bae5282106b5eda13884c6c8780a19e9dcb9904d adds the use of `PostEditorAnalyticsSession` to StoryComposerActivity

#### Note
- for internal events while composing (emoji, text, etc.) we'll gather information in a subsequent PR. This PR tackles the WPAndroid integration side of things only.

#### To test:
- 4b1eaf4752 create a story, then tap on the Upload icon which opens the media picker, then tap on the camera button: observe the tracker gets called with value MEDIA_PICKER_OPEN_WP_STORIES_CAPTURE
- use all of the functionality, verify the code provided in the commits listed above is run

PR submission checklist:
